File: frontend/src/types/activity.ts
export type ActivityEvent = {
  id: string;
  company_id: string;
  occurred_at: string;
  category: "agent" | "approval" | "provider" | "email" | "system" | string;
  source: string;
  action: string;
  title: string;
  summary: string;
  status: string;
  severity: "info" | "warning" | "error" | string;
  actor_display: string;
  entity_type: string;
  entity_id: string | null;
  safe_details: Record<string, string | number | boolean | null>;
  correlation_id: string | null;
};

export type ActivityEventList = {
  items: ActivityEvent[];
  total: number;
  limit: number;
  offset: number;
};

export type ActivityFilters = {
  source?: string;
  severity?: string;
  actor?: string;
  event_type?: string;
  limit?: number;
  offset?: number;
};

File: backend/app/schemas/activity.py
"""Normalized read-only company activity schemas."""

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, Field


class ActivityEventResponse(BaseModel):
    """Safe normalized event for operational timelines."""

    id: UUID
    company_id: UUID
    occurred_at: datetime
    category: str
    source: str
    action: str
    title: str
    summary: str
    status: str
    severity: str
    actor_display: str
    entity_type: str
    entity_id: UUID | None
    safe_details: dict[str, str | int | float | bool | None] = Field(default_factory=dict)
    correlation_id: str | None = None


class ActivityEventListResponse(BaseModel):
    """Paginated normalized company activity."""

    items: list[ActivityEventResponse]
    total: int
    limit: int
    offset: int
